src/servers/graphql-server.ts

```diff
@@ -69,43 +78,43 @@ const sessionContext = ({
   ip,
   body,
 }: {
-  tokenPayload: jwt.JwtPayload | null
+  tokenPayload: jsonwebtoken.JwtPayload
   ip: IpAddress | undefined
   body: unknown
 }): Promise<GraphQLContext> => {
-  const userId = tokenPayload?.uid ?? null
-
-  // TODO move from crypto.randomUUID() to a Jaeger standard
-  const logger = graphqlLogger.child({ tokenPayload, id: crypto.randomUUID(), body })
+  const logger = graphqlLogger.child({ tokenPayload, body })
 
   let domainUser: User | null = null
   let domainAccount: Account | undefined
+
+  // note: value should match (ie: "anon") if not an accountId
+  // settings from dev/ory/oathkeeper.yml/authenticator/anonymous/config/subjet
+  const maybeAid = checkedToAccountId(tokenPayload.sub || "")
+
   return addAttributesToCurrentSpanAndPropagate(
     {
-      [SemanticAttributes.ENDUSER_ID]: userId,
+      [SemanticAttributes.ENDUSER_ID]: tokenPayload.sub,
       [SemanticAttributes.HTTP_CLIENT_IP]: ip,
     },
     async () => {
-      if (userId) {
+      if (!(maybeAid instanceof InvalidAccountIdError)) {
+        const userId = maybeAid as string as UserId // FIXME: fix until User is attached to kratos
         const loggedInUser = await Users.getUserForLogin({ userId, ip, logger })
         if (loggedInUser instanceof Error)
           throw new ApolloError("Invalid user authentication", "INVALID_AUTHENTICATION", {
             reason: loggedInUser,
           })
         domainUser = loggedInUser
 
-        const loggedInDomainAccount = await Accounts.getAccount(
-          domainUser.defaultAccountId,
-        )
+        const loggedInDomainAccount = await Accounts.getAccount(maybeAid)
         if (loggedInDomainAccount instanceof Error) throw Error
         domainAccount = loggedInDomainAccount
-      }
 
-      addAttributesToCurrentSpan({ [ACCOUNT_USERNAME]: domainAccount?.username })
+        addAttributesToCurrentSpan({ [ACCOUNT_USERNAME]: domainAccount?.username })
+      }
 
       return {
         logger,
-        uid: userId,
```

Comment:
I cant see modifications in queries and mutations. We dont have TS in graphql layer so you need to search `uid` and update all queries and mutations that use `uid`.

Reply:
there are only 2 and they are in the admin api. I think they are passed explicitly as an argument as opposed to be deduced from the context. can you double check?

Reply:
yep, you are right. they are passed as parameters.